### DISCUSSION TITLE
Unable to download-artifacts

### DISCUSSION THREAD

**Discussion by rafaels-dev** [community/community#26618] [Actions]
<p>I’m creating a workflow that need to save artifacts in build step and same artifacts in deploy step.<br>
upload-artifacts working well, but download-artifacts can’t find the folder I’ve uploaded.</p>
<p>upload-artifact</p>
<pre><code class="lang-auto">    - name: Save public folder
      uses: actions/upload-artifact@v2
      with:
        name: public
        path: public
</code></pre>
<p>Upload Log:</p>
<pre><code class="lang-auto">  Save public folder2s
Artifact public has been successfully uploaded!
Run actions/upload-artifact@v2
With the provided path, there will be 36 files uploaded
Total size of all the files uploaded is 798442 bytes
Finished uploading artifact public. Reported size is 798442 bytes. There were 0 items that failed to upload
Artifact public has been successfully uploaded!
</code></pre>
<p>Download artifact’s folder:</p>
<pre><code class="lang-auto">- name: Download files to publish
        uses: actions/download-artifact@v2
        with:
          name: public
</code></pre>
<p>Download log</p>
<pre><code class="lang-auto">Download files to publish0s
##[error]Unable to find any artifacts for the associated workflow
Run actions/download-artifact@v2
##[error]Unable to find any artifacts for the associated workflow
</code></pre>
<p>This is the workflow file: <a href="https://github.com/rafaels-dev/gatsby-github-personal-website/blob/bd66d34420fdc954b7e6289b4ecaa8f2e9b29cdf/.github/workflows/node.js.yml" rel="nofollow noopener">https://github.com/rafaels-dev/gatsby-github-personal-website/blob/bd66d34420fdc954b7e6289b4ecaa8f2e9b29cdf/.github/workflows/node.js.yml</a></p>
<p>You can see the error in this run <a href="https://github.com/rafaels-dev/gatsby-github-personal-website/actions/runs/122591922" rel="nofollow noopener">https://github.com/rafaels-dev/gatsby-github-personal-website/actions/runs/122591922</a></p>
<p>What’s wrong on my file?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I notice your “deploy” job doesn’t have a <code>needs: build</code> setting, so it should be started in parallel (if there’s room in the runner queue). Considering the build is likely to take some time the likely answer is that the artifact hasn’t been uploaded yet when the <code>download-artifact</code> step is running.</p>

**Reply by epignatelli** [reply]
But that defeats the entire purpose of `needs:`, no?
The job should be considered completed when the artifact is uploadaed, not when the upload command is conveyed.

**Reply by airtower-luna** [reply]
How do you mean? The problem in the original question was that there was no `needs`, so nothing tried to wait. With the `needs` relationship using artifacts from the needed job should just work.

**Reply by epignatelli** [reply]
Sorry -- I completely misunderstood the question AND the answer as well -- got it.

**Comment by rafaels-dev** [comment]
<p>Thank you. It worked fine!</p>

**Comment by oppknox** [comment]
<p>what was the actual command you ended up using? After reading about “needs” I’m still not sure how to direct it towards my build.</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/actions/reference/workflow-syntax-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Reply by tukhi** [reply]
I have a similar case:
- 2 jobs within same workflow
- job1 triggered when pushed to branch
- it uploads artifact from job1 and it shows successfully uploaded. Link to artifact also appears in GH Actions. I can download a zipped copy to my local machine. 
- Job2 has a different trigger (PR closed/merged with main). when I trigger job2, download cannot find the artifact. 
- If I use "needs", it reruns job1, which I do not want to be re-triggered. Retriggering creates an unwanted new artifact.
- I have not been able to resolve this issue.

**Reply by ianrussel** [reply]
@tukhi I have same issue. Did you solved it?

**Reply by ggabijaa** [reply]
@tukhi did you resolve it? having the same issue

**Comment by Binary-Ninja-007** [comment]
From the information provided, it appears that the issue lies with the download-artifact step. The error message suggests that no artifacts were found for the associated workflow.

Here are a few things you can check to troubleshoot the problem:

Verify Artifact Name: Ensure that the artifact name specified in the download-artifact step (name: public) matches the name used in the upload-artifact step (name: public).

Check Workflow Context: Make sure that the download-artifact step is declared within the same workflow or across dependent workflows that have access to the artifacts.

Artifact Retention Policy: Confirm that the artifacts are not being automatically deleted due to a retention policy set either within the workflow or in the repository settings. If the artifacts are automatically deleted, they won't be available for download.

Workflow Execution: Check if the workflow execution that triggers the download-artifact step is related to the workflow that performed the upload-artifact step. Artifacts are only available within the context of the same workflow run.

Double-checking these points should help you identify any potential issues and make the necessary adjustments. Good luck! 😊

**Comment by peternied** [comment]
Since this came up as a top search result:

We accepted a dependabot update [1] for `download-artifact` and we did not update `upload-artifact` which caused our artifacts not to be available.  Update both dependencies at the same time to resolve this issue, as was done with [2].

Related:
- [1] https://github.com/opensearch-project/OpenSearch/pull/11631
- [2] https://github.com/opensearch-project/OpenSearch/pull/11647
